expression/builtin_string_test.go

```diff
@@ -984,6 +984,31 @@ func (s *testEvaluatorSuite) TestLpad(c *C) {
 	}
 }
 
+func (s *testEvaluatorSuite) TestQuote(c *C) {
+	defer testleak.AfterTest(c)()
+
+	tbl := []struct {
+		arg interface{}
+		ret interface{}
+	}{
+		{`Don\'t!`, `'Don\'t!'`},
```

Comment:
See also the comment here https://github.com/pingcap/tidb/pull/2923  @ariesdevil 

The input of `eval` function is not exactly what you write down, lexer will escape them.